modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/rule/AbstractIgniteJoinConverterRule.java

```diff
@@ -82,57 +84,56 @@ protected AbstractIgniteJoinConverterRule(String descriptionPrefix, HintDefiniti
 
     /** */
     private boolean disabledByHints(LogicalJoin join) {
-        if (HintUtils.allRelHints(join).isEmpty())
+        Collection<TableScan> joinTables = joinTables(join);
+
+        Collection<RelHint> hints = Stream.concat(
+            F.flatCollections(joinTables.stream().map(TableScan::getHints).collect(Collectors.toList())).stream(),
+            HintUtils.allRelHints(join).stream()).collect(Collectors.toList());
```

Comment:
A lot of streams and additional lists created here. Let's simplify this to something like:
```
        Collection<RelHint> hints = new ArrayList<>();
        joinTables.forEach(t -> hints.addAll(t.getHints()));
        hints.addAll(HintUtils.allRelHints(join));
```
The same 3 lines, but much easier to understand and without a lot of garbage creation.